compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1485,6 +1485,10 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         self.probe(|_| {
             // First check that the self type can be related.
             let sub_obligations = match self.at(&ObligationCause::dummy(), self.param_env).sup(
+                // https://github.com/rust-lang/rust/issues/116652
+                // Failed tests:
+                // tests/ui/impl-trait/issues/issue-70877.rs
+                // tests/ui/type-alias-impl-trait/type-alias-impl-trait.rs
```

Comment:
We may want to mention that these 2 are ICEs and describe them here?

Reply:
@lqd Thank you for your review. Yes, I've reverted and made a new commit